osf/models/preprint_provider.py

```diff
@@ -79,3 +83,7 @@ def banner_path(self):
             return '/static/img/preprint_providers/{}'.format(self.logo_name)
         else:
             return None
+
+    @property
+    def metaschema(self):
+        return MetaSchema.objects.get(name='Preprint Provider')
```

Comment:
Ah, they currently aren't linked at all. 

I think I need to add a validator to the `extra` field that should call the `MetaSchema` class' [`validate_metadata`](https://github.com/CenterForOpenScience/osf.io/blob/develop/osf/models/metaschema.py#L69).

It doesn't appear that [`validate_metadata`](https://github.com/CenterForOpenScience/osf.io/blob/develop/osf/models/metaschema.py#L69) is tested at all, so I'll add tests for that as well.